config/schema/3.0/patches/db_schema_patch-3.1.sql

```diff
@@ -164,4 +164,28 @@ VALUES("Occurrence Associations Type", "omoccurassociations", "relationship", "a
 INSERT INTO ctcontrolvocabterm(cvID, term, termDisplay)
 SELECT cvID, "fieldNotes", "Field Notes" FROM ctcontrolvocab WHERE tableName = "omoccurassociations" AND fieldName = "relationship" AND filterVariable = "associationType:resource";
 INSERT INTO ctcontrolvocabterm(cvID, term, termDisplay)
-SELECT cvID, "genericResource", "Generic Resource" FROM ctcontrolvocab WHERE tableName = "omoccurassociations" AND fieldName = "relationship" AND filterVariable = "associationType:resource";
\ No newline at end of file
+SELECT cvID, "genericResource", "Generic Resource" FROM ctcontrolvocab WHERE tableName = "omoccurassociations" AND fieldName = "relationship" AND filterVariable = "associationType:resource";
+
+# Establish a table to track third party auth
+
+CREATE TABLE `users_thirdpartyauth` (
+  `id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+  `uid` INT(10) UNSIGNED NOT NULL,
+  `sub_uuid` VARCHAR(100) NOT NULL,
+  `provider` VARCHAR(200) NOT NULL,
+  PRIMARY KEY (`id`),
+  CONSTRAINT `fk_users_uid`
+    FOREIGN KEY (`uid`)
+    REFERENCES `users` (`uid`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE);
+
+# Clean up localitySecurity for occurrences that are cultivated and have not explicitly had their localitySecurity edited to be 1 (and are missing a security reason) more recently than it has been edited to 0.
+
+UPDATE omoccurrences o INNER JOIN omoccuredits e ON o.occid = e.occid
+LEFT JOIN (SELECT occid, ocedid FROM omoccuredits WHERE fieldName = "localitySecurity" AND fieldValueNew = 0) e2 ON e.occid = e2.occid AND e.ocedid < e2.ocedid
+SET o.localitySecurity = 1, o.localitySecurityReason = "[Security Setting Explicitly Locked]"
+WHERE o.localitySecurityReason IS NULL AND e.fieldName = "localitySecurity" AND e.fieldValueNew = 1
+AND e2.occid IS NULL;
+
+UPDATE omoccurrences SET localitySecurity=0 WHERE cultivationStatus=1 AND localitySecurity=1 AND localitySecurityReason IS NULL;
```

Comment:
@Atticus29 Are we sure all portals want cultivated occurrences of sensitive species to be geolocated?

Reply:
@GregPost-ASU I can't think of any reason why not. These statements should ignore any occurrences that a user has explicitly set localitySecurity on. @egbot?